### PR TITLE
config: Create config for BigTreeTech Octopus

### PR DESCRIPTION
**Pull request by charlespick**
Information and some lines verbatim were sourced from the official BigTreeTech Octopus
GitHub repository to make this configuration as well as existing configurations in the same
folder as this one

All commits before this pull request was opened were...
Signed-off-by: Charles Pickering <charles.pickering@live.com>

### CODE REVIEW

**Comment by batmanreturns1** [comment]
I'm having an issue with my bed temp. Keeps failing to heat when it gets to say 50/60 (I'm doing tests at up to 100 mainly). I've been told the issue is around the 2 degrees per 60 seconds seconds with a +/-5 of the target. I've attempted to change this in my printer.cfg but sadly this just gives me further error codes🤷🏻‍♂️

I'm using a Bigtreetech octopus V1.1 with 2209s and an OEM CR-10 S5. 

Please feel free to ask any further questions sorry if I've missed key information. 

Thanks in advance for your help!

**Comment by charlespick** [comment]
@batmanreturns1 please open an issue if you believe you have diagnosed an issue with the software. For general assistance with using the software please seek help on Discord or the forum.

**Comment by KevinOConnor** [comment]
Thanks.  Change is fine, but github is showing a merge conflict.  Not sure why, but probably needs a `git rebase`.

-Kevin

**Comment by charlespick** [comment]
Says it's ready to merge here. I'll check on my computer tomorrow.

**Comment by charlespick** [comment]
GitHub says there's no conflicts with the base branch. I merged master into here after that typo in the test dictionaries was fixed.

File: config/generic-bigtreetech-octopus.cfg [hunk 1]
```diff
@@ -0,0 +1,294 @@
+# This file contains common pin mappings for the BigTreeTech Octopus.
+# To use this config, the firmware should be compiled for the
+# STM32F466 with a "32KiB bootloader" and a "12MHz crystal" clock reference.
```
Comment (a7hybnj2) [inline config/generic-bigtreetech-octopus.cfg:3]: Did you mean STM32F446?

**Comment by KevinOConnor** [comment]
Ah - github was just showing me confusing info.  I merged this change along with a fix for the mcu typo on top (commit 4c6b790d).

Thanks.
-Kevin

**Comment by charlespick** [comment]
> Thanks.  Change is fine, but github is showing a merge conflict.  Not sure why, but probably needs a `git rebase`.
> 
> 
> 
> -Kevin

So when I rebased last night to get #445 mergable GitHub rejected my changes and I ended up just creating a merge commit. How are you supposed to use rebase?

**Comment by KevinOConnor** [comment]
If you do a web search for "git rebase" you'll find documentation (for example - https://git-scm.com/book/en/v2/Git-Branching-Rebasing ).  When I merge in changes, I always "rebase" those changes onto the current master branch, and the Klipper master branch is itself never rebase'd (so that it has a stable history).

-Kevin

**Comment by KevinOConnor** [comment]
Oh - if github was warning about pushing a branch up that has been rebase'd, then yes, it is necessary to use a `git push --force` to push a branch that has been rebase'd.  I do that when updating any of my "work" branches (but again, I never rebase the Klipper master branch).

-Kevin

**Comment by charlespick** [comment]
I thought force pushing should be avoided because it screws up peoples' local repositories

**Comment by KevinOConnor** [comment]
Right - which is why I never force push the Klipper master branch.  It's a common strategy to rebase and force push development branches however.

-Kevin

**Pull request merged**